clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -715,45 +717,32 @@ scheduler_tmpl_t<RecordType, ReaderType>::advance_region_of_interest(int output_
         cur_range = input.regions_of_interest[input.cur_region];
     }
 
-    if (cur_instr >= cur_range.start_instruction)
+    // If we're within one and already skipped, just exit to avoid re-requesting a skip
+    // and making no progress (we're on the inserted timetamp + cpuid and our cur instr
+    // count isn't yet the target).
+    if (input.in_cur_region && cur_instr >= cur_range.start_instruction - 1)
```

Comment:
If the first roi starts at instr 0, then in _cur_region may be false (if it is initialized to false) but we're already in the current region. In that case, should we be returning here?

Also see comment above about overlapping rois.